pkgs/development/python-modules/z3-solver/default.nix

```diff
@@ -0,0 +1,29 @@
+{ buildPythonPackage
+, fetchPypi
+, lib
+, setuptools
+}:
+
+buildPythonPackage rec {
+  pname = "z3-solver";
```

Comment:
I think z3 is already a python package as ``pythonPackages.z3``
https://github.com/NixOS/nixpkgs/blob/acdd2395784f0aae02f9eb364071aeda00f199cd/pkgs/top-level/python-packages.nix#L7226-L7228